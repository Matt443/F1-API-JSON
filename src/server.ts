import { getDriverData } from "./scraper/driver-data";
import { getTeamsData } from "./scraper/team-data";
import { getDriverStandings } from "./scraper/driver-standings";
import { getConstructorStandings } from './scraper/constructors-standings';
import { getWorldChampions } from './scraper/world-champions'

export {
    getDriverData,
    getTeamsData,
    getDriverStandings,
    getConstructorStandings,
    getWorldChampions
}