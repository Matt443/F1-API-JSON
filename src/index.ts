import { getRaceResults } from "./server";

async function test() {
    await getRaceResults();
    console.log();
}
test();
