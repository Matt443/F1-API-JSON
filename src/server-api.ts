import express from "express";
import cors from "cors";
import helmet from "helmet";
import dotenv from "dotenv";

import router from "./endpoints-api/routes-api";
import { getFullRaceResults, getRaceResults } from "./server";

dotenv.config();

const app = express();

app.use(cors());
app.use(helmet());

app.use("/", router);

app.listen(process.env.PORT || 4005, () => {
    console.log(`server is running on ${process.env.PORT || 4005}`);
}).on("error", (error) => {
    throw new Error(error.message);
});
