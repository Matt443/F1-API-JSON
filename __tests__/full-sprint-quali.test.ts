import { getFullSprintShootout } from "../src/server";

test("get race results", async () => {
    expect(await getFullSprintShootout(2024)).toMatchSnapshot();
});
