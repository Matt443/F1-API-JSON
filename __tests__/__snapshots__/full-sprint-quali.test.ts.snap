// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`get race results 1`] = `
[
  {
    "code": "PIA",
    "laps": 16,
    "name": "Oscar Piastri",
    "number": 81,
    "position": "1",
    "team": "McLaren Mercedes",
    "times": {
      "q1": "1:10.265",
      "q2": "1:09.239",
      "q3": "1:08.899",
    },
  },
  {
    "code": "NOR",
    "laps": 14,
    "name": "Lando Norris",
    "number": 4,
    "position": "2",
    "team": "McLaren Mercedes",
    "times": {
      "q1": "1:09.477",
      "q2": "1:09.063",
      "q3": "1:08.928",
    },
  },
  {
    "code": "LEC",
    "laps": 18,
    "name": "Charles Leclerc",
    "number": 16,
    "position": "3",
    "team": "Ferrari",
    "times": {
      "q1": "1:10.388",
      "q2": "1:09.248",
      "q3": "1:09.153",
    },
  },
  {
    "code": "VER",
    "laps": 12,
    "name": "Max Verstappen",
    "number": 1,
    "position": "4",
    "team": "Red Bull Racing Honda RBPT",
    "times": {
      "q1": "1:10.409",
      "q2": "1:09.489",
      "q3": "1:09.219",
    },
  },
  {
    "code": "SAI",
    "laps": 15,
    "name": "Carlos Sainz",
    "number": 55,
    "position": "5",
    "team": "Ferrari",
    "times": {
      "q1": "1:10.503",
      "q2": "1:09.500",
      "q3": "1:09.257",
    },
  },
  {
    "code": "RUS",
    "laps": 15,
    "name": "George Russell",
    "number": 63,
    "position": "6",
    "team": "Mercedes",
    "times": {
      "q1": "1:10.479",
      "q2": "1:09.683",
      "q3": "1:09.443",
    },
  },
  {
    "code": "GAS",
    "laps": 15,
    "name": "Pierre Gasly",
    "number": 10,
    "position": "7",
    "team": "Alpine Renault",
    "times": {
      "q1": "1:10.630",
      "q2": "1:09.610",
      "q3": "1:09.622",
    },
  },
  {
    "code": "LAW",
    "laps": 11,
    "name": "Liam Lawson",
    "number": 30,
    "position": "8",
    "team": "RB Honda RBPT",
    "times": {
      "q1": "1:10.576",
      "q2": "1:09.827",
      "q3": "1:09.941",
    },
  },
  {
    "code": "ALB",
    "laps": 12,
    "name": "Alexander Albon",
    "number": 23,
    "position": "9",
    "team": "Williams Mercedes",
    "times": {
      "q1": "1:10.366",
      "q2": "1:09.844",
      "q3": "1:10.078",
    },
  },
  {
    "code": "BEA",
    "laps": 12,
    "name": "Oliver Bearman",
    "number": 50,
    "position": "10",
    "team": "Haas Ferrari",
    "times": {
      "q1": "1:10.442",
      "q2": "1:09.629",
      "q3": "DNF",
    },
  },
  {
    "code": "HAM",
    "laps": 13,
    "name": "Lewis Hamilton",
    "number": 44,
    "position": "11",
    "team": "Mercedes",
    "times": {
      "q1": "1:10.625",
      "q2": "1:09.941",
    },
  },
  {
    "code": "HUL",
    "laps": 8,
    "name": "Nico Hulkenberg",
    "number": 27,
    "position": "12",
    "team": "Haas Ferrari",
    "times": {
      "q1": "1:10.466",
      "q2": "1:09.964",
    },
  },
  {
    "code": "PER",
    "laps": 10,
    "name": "Sergio Perez",
    "number": 11,
    "position": "13",
    "team": "Red Bull Racing Honda RBPT",
    "times": {
      "q1": "1:10.392",
      "q2": "1:10.024",
    },
  },
  {
    "code": "COL",
    "laps": 9,
    "name": "Franco Colapinto",
    "number": 43,
    "position": "14",
    "team": "Williams Mercedes",
    "times": {
      "q1": "1:10.470",
      "q2": "1:10.275",
    },
  },
  {
    "code": "BOT",
    "laps": 12,
    "name": "Valtteri Bottas",
    "number": 77,
    "position": "15",
    "team": "Kick Sauber Ferrari",
    "times": {
      "q1": "1:10.861",
      "q2": "1:10.595",
    },
  },
  {
    "code": "ALO",
    "laps": 6,
    "name": "Fernando Alonso",
    "number": 14,
    "position": "16",
    "team": "Aston Martin Aramco Mercedes",
    "times": {
      "q1": "1:10.978",
    },
  },
  {
    "code": "OCO",
    "laps": 6,
    "name": "Esteban Ocon",
    "number": 31,
    "position": "17",
    "team": "Alpine Renault",
    "times": {
      "q1": "1:11.052",
    },
  },
  {
    "code": "TSU",
    "laps": 6,
    "name": "Yuki Tsunoda",
    "number": 22,
    "position": "18",
    "team": "RB Honda RBPT",
    "times": {
      "q1": "1:11.121",
    },
  },
  {
    "code": "STR",
    "laps": 6,
    "name": "Lance Stroll",
    "number": 18,
    "position": "19",
    "team": "Aston Martin Aramco Mercedes",
    "times": {
      "q1": "1:11.280",
    },
  },
  {
    "code": "ZHO",
    "laps": 5,
    "name": "Zhou Guanyu",
    "number": 24,
    "position": "20",
    "team": "Kick Sauber Ferrari",
    "times": {
      "q1": "1:12.978",
    },
  },
]
`;
